Sources/NChatCore/Model/ChatUserData.swift

```diff
@@ -0,0 +1,123 @@
+// Copyright © 2022 Nomasystems S.L. All rights reserved.
+
+import Foundation
+import XMPPCore
+
+/**
+ Defines data of the particular user connecting to the chat.
+
+ All fields are optional. Those marked as metadata will only be used to be sent when joining the chat.
+ */
+public struct ChatUserData {
+    /// Attachments that the user might send during the chat.
+    public let attachments: [ChatAttachment]
+    /// Country code of the user. Metadata.
+    public let countryCode: String?
+    /// Email of the user. Metadata.
+    public let email: String?
+    /// Name of the user. Metadata.
+    public let name: String?
+    /// Location (e.g. view) from where the user opened the chat. Metadata.
+    public let location: String?
+    /// Identifier of the user's product. Metadata.
+    public let productSku: String?
+    /// Identifier of the user's order. Metadata.
+    public let orderId: String?
+    /// Topic of the chat
```

Comment:
This was already present but we might as well correct it.
```suggestion
    /// Topic of the chat. Metadata.
```